doboto/Droplet.py

```diff
@@ -145,32 +147,30 @@ def shutdown(self, droplet_id, with_tag=None):
         Shutdown a droplet, or tagged droplets
         """
         if with_tag is not None:
-            uri = "%s/actions?tag_name=%s" % (self.uri, with_tag)
+            uri = "{}/actions?tag_name={}".format(self.uri, with_tag)
         else:
-            uri = "%s/%s/actions" % (self.uri, droplet_id)
+            uri = "{}/{}/actions".format(self.uri, droplet_id)
         attribs = {"type": "shutdown"}
         return self.make_request(uri, 'POST', attribs=attribs)
 
     def restore(self, droplet_id, image):
         """
         Restore a droplet from an backup image
         """
-        uri = "%s/%s/actions" % (self.uri, droplet_id)
-
-        try:
-            image = int(image)
-        except ValueError:
-            pass
+        uri = "{}/{}/actions".format(self.uri, droplet_id)
 
-        attribs = {"type": "restore", "image": image}
+        if isinstance(image, int):
+            attribs = {"type": "restore", "image": image}
```

Comment:
from the v2 api docs https://developers.digitalocean.com/documentation/v2/#restore-a-droplet `image: string if an image slug. number if an image ID.` So this the goal here is to build the attributes such that if the user passes an `int` it will be retained as such, where as if a string is passed its assumed to be a slug name.